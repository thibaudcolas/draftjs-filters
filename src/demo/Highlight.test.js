import React from "react"
import { shallow } from "enzyme"
import Highlight from "./Highlight"

describe("Highlight", () => {
  it("renders", () => {
    expect(
      shallow(<Highlight value="" language="javascript" />),
    ).toMatchSnapshot()
  })

  it("onCopy", () => {
    document.execCommand = jest.fn()
    const wrapper = shallow(<Highlight value="" language="javascript" />)

    wrapper.find("button").simulate("click")

    expect(document.execCommand).toHaveBeenCalledWith("copy")
  })
})
