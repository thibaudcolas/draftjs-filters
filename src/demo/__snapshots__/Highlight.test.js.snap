// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight renders 1`] = `
<pre
  className="language-javascript"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "position": "absolute",
        "right": "1rem",
      }
    }
  >
    Copy
  </button>
  <code
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</pre>
`;
